src/core/struct_tree.js

```diff
@@ -67,6 +69,439 @@ class StructTreeRoot {
       this.roleMap.set(key, value.name);
     });
   }
+
+  static async createStructureTree(
```

Comment:
*Another general suggestion, given the sheer number of parameters that many of the new functions require:*
Would it make sense to pass parameter-Objects to all these methods instead, since there should be much less risk of breaking something if we accidentally mess up the ordering?